src/mta-in/tests/test_handler.py

```
@@ -0,0 +1,246 @@
+async def test_data_oversize_bumps_soft_errors():
+    session, envelope = _session(), _envelope()
+    envelope.content = b"x" * (settings.MAX_INCOMING_EMAIL_SIZE + 10)
+    reply = await _handler().handle_DATA(None, session, envelope)
+    assert reply.startswith("552")
+    assert getattr(session, _SOFT_ERRORS_ATTR) == 1
+
+
+@pytest.mark.asyncio
+async def test_data_max_envelopes_bumps_soft_errors():
+    session, envelope = _session(), _envelope()
+    setattr(session, "_pymta_envelopes", settings.PYMTA_MAX_ENVELOPES_PER_CONNECTION)
```

Ruff (0.15.20)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)